plotly/tools.py

```diff
@@ -1443,6 +1443,376 @@ class FigureFactory(object):
     """
 
     @staticmethod
+    def create_scatterplotmatrix(df, useindex = False, index = " ",
+                                 diagonal = "Scatter", symbol = 0, size = 6,
+                                 height = " ", width = " ", jitter = 0,
+                                 title = "Scatterplot Matrix"):
+
+        from plotly import tools
+        import plotly.plotly as py
+        import plotly.graph_objs as go
+        import pandas as pd
+        import random as r
+       
+        colors = ["rgb(31, 119, 180)", "rgb(255, 127, 14)",
+                  "rgb(44, 160, 44)", "rgb(214, 39, 40)",
+                  "rgb(148, 103, 189)", "rgb(140, 86, 75)",
+                  "rgb(227, 119, 194)", "rgb(127, 127, 127)",
+                  "rgb(188, 189, 34)", "rgb(23, 190, 207)"]
+
+        matrix = []
+        col_names = []
+        index_vals = []
+        diagonal_choices = ["Scatter", "Histogram"]
+
+        # Check if pandas dataframe
+        if type(df) != pd.core.frame.DataFrame:
+            raise exceptions.PlotlyError("Dataframe not inputed. Please " 
+                                         "use a pandas dataframe to pro" 
+                                         "duce a scatterplot matrix.")
+
+        # Check if dataframe is 1 row or less.
+        if len(df.columns) <= 1:
+            raise exceptions.PlotlyError("Dataframe has only one column. To "
+                                         "use the scatterplot matrix, use at "
+                                         "least 2 columns.")
+
+        # Check that diagonal parameter is selected properly
+        if diagonal not in diagonal_choices:
+            raise exceptions.PlotlyError("Make sure diagonal is set to "
+                                         "either Scatter or Histogram.")
```

Comment:
Can you condense all the input checks into a single private `_` function to reduce the size of this method? It's hard to consider in it's entirety with so many lines (and hard to test!).

Here's an [example](https://github.com/plotly/plotly.py/blob/master/plotly/tools.py#L1501)

Note that you might be able to reuse other validation methods in `FigureFactory`.